docs/developer-guide/typeConversions.md

```diff
@@ -0,0 +1,211 @@
+
+## Data conversions
+
+There are defined conversions from all known available types to all others.
+
+### Conversion from Double
+
+- INT -> trunc(val)
+- DOUBLE -> val
+- COMPLEX -> val+0j
+- VECTOR - > [val]
+- COMPLEX_VECTOR -> [val+0j]
+- NAMED_POINT ->{"value", val}
+- STRING -> string representation such that all required bits are included
+- BOOL ->(val!=0)?"1":"0"
+
+### Conversion from INT
+
+- INT -> val
+- DOUBLE -> val [^1]
+- COMPLEX -> val+0j
+- VECTOR - > [val]
+- COMPLEX_VECTOR -> [val+0j]
+- NAMED_POINT ->{"value", val} [^2]
+- STRING -> std::to_string(val)
+- BOOL ->(val!=0)?"1":"0"
+
+[^1]: for conversion to double lossless only if value actually fits in a double mantissa value.
+[^2]: For a named point conversion. If the value doesn't fit in double the string translation is placed in the string field and a NAN value in the value segment to ensure no data loss.
+
+### Conversion from String
+
+- INT -> `getIntFromString(val)`
+- DOUBLE -> `getDoubleFromString(val)`
+- COMPLEX -> `getComplexFromString(val)`
+- VECTOR - > `helicsGetVector(val)`
+- COMPLEX_VECTOR -> `helicsGetComplexVector(val)`
+- NAMED_POINT ->{val, NAN}
+- STRING -> val
+- BOOL ->(helicsBoolValue(val))?"1":"0"
+
+#### helicsGetComplexVector
+
+This method will read a vector of numbers from a string in either JSON format, or `c[X1,X2,...XN]` where XN is a complex number of the format R+Ij.
+It can also interpret `v[X1,X2, ..., XN]` in which case the values are assumed to be alternating real and imaginary values. If the string is a single value either real or complex it is placed in a vector of length 1.
+
+#### helicsGetVector
+
+This function is similar to `helicsGetComplexVector` with distinction that string like `v[X1,X2, ..., XN]` are all assumed separate values, and complex vectors are generated as alternating real and imaginary values.
+
+#### getIntFromString
+
+Converts a string into 64 bit integer. If the string has properties of a double or vector it will truncate the values from `getDoubleFromString`.
+
+#### getDoubleFromString
+
+Converts a string into a double value, a complex, or a vector of real or complex numbers. If the vector is more than a single element the output is the vector norm of the vector. If the string is not convertible the invalid_double is returned (-1e49).
+
+#### helicsGetComplexFromString
+
+Similar to getDoubleFromString in conversion of vectors. It will convert most representations of complex number patterns using a trailing `i` or `j` for the imaginary component and assumes the imaginary component is last. The real component can be omitted if not present. For Example `4.7+2.7j`or`99.453i`
```

Comment:
```suggestion
Similar to getDoubleFromString in conversion of vectors. It will convert most representations of complex number patterns using a trailing `i` or `j` for the imaginary component and assumes the imaginary component is last. The real component can be omitted if not present, for example `4.7+2.7j`or`99.453i`
```